lib/compile.coffee

```diff
@@ -199,9 +200,11 @@ compile = (options) ->
         # Strip out any filters (e.g. ng-repeat="foo in bar | limitTo: 10")
         # and split by whitespace and compact the result (remove any empty
         # strings in the list) as well as the 'track by' option in angular
-        repeatExpSplit = _.compact repeatExp
+        repeatExp = _.compact repeatExp if typeof repeatExp isnt 'string'
+
+        repeatExpSplit = repeatExp
           .split(' | ')[0]
-          .split('track by')[0]
+          .split('track by')[0].trim()
```

Comment:
There was an extra space at the end when you use `track by` in an ng-repeat that was causing compylr to fail.